runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityPlugin.java

```diff
@@ -421,4 +423,39 @@ private void onTileObject(Tile tile, TileObject oldObject, TileObject newObject)
 			}
 		}
 	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (config.showShortcutLevel())
+		{
+			MenuEntry[] entries = client.getMenuEntries();
+
+			//Skip anything where there's 0 chance it's a agi shortcut.
+			if (event.getTarget() == null || event.getTarget() == ""){
+				return;
+			}
+
+			Map<TileObject,Obstacle> obstacles = getObstacles();
+			for (Obstacle o : obstacles.values())
+			{
+				//Compare the event ID to nearby obstacle ID's
+				for (int obId: o.getShortcut().getObstacleIds()){
+					if (obId == event.getIdentifier()){
+						AgilityShortcut agilityShortcut = o.getShortcut();
+						for (MenuEntry entry : entries) {
+							//Check it's the non-examine
```

Comment:
I believe you can use `MenuAction.EXAMINE` to compare against. Also, this comment isn't correct since it doesn't explain the comparison of the entry target to the event target nor the check for level text. Generally, comments which are explaining what the code is doing as opposed to why it's being done should be removed. Much of this code can be made more clear by being broken into smaller statements, using better variable naming, etc.

Reply:
Ah I hadn't noticed those. I've added a guard case for some which I imagine are more common.